### PR TITLE
AB#148 implement attack command

### PR DESCRIPTION
**Pull request by sir-marc**

### CODE REVIEW

**Comment by StuckiSimon** [comment]
As the new game engine changes the way we exchange objects within the user execution context it makes sense to integrate the changes ahead of approving this PR.
Could you update your branch with the newest changes from `master`?

File: game-backend/src/game-engine/intent/AttackIntent.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    const yDiff = Math.abs(unit.position.y - targetUnit.position.y);
+    const diff = Math.sqrt(xDiff ** 2 + yDiff ** 2);
+
+    return diff <= ATTACK_RANGE;
+  }
+
+  private static attack(targetUnit: Unit, state: State): State {
+    return produce(state, draft => {
+      const defendingUnit = getUnit(draft, targetUnit.id);
+      defendingUnit.health -= HIT_DAMAGE;
```
Comment (StuckiSimon) [inline game-backend/src/game-engine/intent/AttackIntent.ts:32]: Given:
- an enemy unit with 5 health is on the **left side** of your own unit
- you attack the enemy unit
- the enemy unit has -5 health and is therefore dead. It shouldn't be on the field anymore
- the own unit wants to move **left** - this is not possible as the dead enemy unit is still on the field

Or do I miss something?

**Review by StuckiSimon — COMMENTED** [review]
There are merge conflicts and as far as I'm aware of there might be a flaw in the execute attack logic.

File: game-backend/src/game-engine/intent/AttackIntent.ts (hunk 1 above)
Comment (michaelschufi) [reply game-backend/src/game-engine/intent/AttackIntent.ts:32]: Yes, you are right. We need to remove the unit like we do when it starves to death. It should be the same dying behaviour after it gets to <= 0 HP.

**Review by StuckiSimon — APPROVED** [review]
Looks good 👍

**Pull request merged**